src/components/PasswordReset/index.jsx

```diff
@@ -0,0 +1,114 @@
+import { Alert, Input } from "antd";
+import React, { useState } from "react";
+import { validateEmail } from "../../functions";
+import "./index.css";
+
+const Login = (props) => {
+  const [values, setValues] = useState({ email: "" });
+
+  const handleChange = (key, value) => {
+    let newValues = Object.assign({}, values);
+    newValues[key] = value;
+    setValues(newValues);
+  };
+  const [errors, setErrors] = useState({});
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    let { email, password } = values;
+    let errors = {};
+    if (!validateEmail(email)) {
+      errors["email"] = "Email format error";
+    }
+    if (email.trim() === "") {
+      errors["email"] = "Please fill this field";
+    }
+
+    setErrors(errors);
+    if (Object.keys(errors).length > 0) {
+      return;
+    }
+    if (typeof props.onSubmit == "function") {
```

Comment:
again you shouldn't have to use typeof here. if you are sometimes getting not a function, its a sign of a bug in your code. i would consider console.logging props and seeing what it is thats causing it to crash

Reply:
this way I'm passing onSubmit prop as an optional prop so it doesn't crash the app if it's not provided. do you need there is a need to update it ?